### PR TITLE
Add functional methods for networks and subnetworks

### PR DESCRIPTION
**Pull request by mziccard**
This PR adds:
- functional methods to `Compute` and `ComputeRpc` for networks and subnetworks
- `Network` and `Subnetwork` functional classes
- Unit tests for new functional methods, `Network` and `Subnetwork`
- Integration tests for networks and subnetworks

### CODE REVIEW

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java [hunk 1]
```diff
@@ -964,6 +964,88 @@ public static DiskFilter notEquals(DiskField field, long value) {
...
+    }
+
+    /**
+     * Returns an equals filter for the given field and string value. For string fields,
+     * {@code value} is interpreted as a regular expression using RE2 syntax. {@code value} must
+     * match the entire field.
+     *
+     * @see <a href="https://github.com/google/re2/wiki/Syntax">RE2</a>
+     */
+    public static NetworkFilter equals(NetworkField field, String value) {
```
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java]: Codacy Issue found: [The method name and parameter number are suspiciously close to equals(Object)](https://www.codacy.com/app/mziccard/gcloud-java/file/2431752990/issues/source?bid=3140968&fileBranchId=3241696#l1018)

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java [hunk 2]
```diff
@@ -964,6 +964,88 @@ public static DiskFilter notEquals(DiskField field, long value) {
...
+    }
+
+    /**
+     * Returns an equals filter for the given field and string value. For string fields,
+     * {@code value} is interpreted as a regular expression using RE2 syntax. {@code value} must
+     * match the entire field.
+     *
+     * @see <a href="https://github.com/google/re2/wiki/Syntax">RE2</a>
+     */
+    public static SubnetworkFilter equals(SubnetworkField field, String value) {
```
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java]: Codacy Issue found: [The method name and parameter number are suspiciously close to equals(Object)](https://www.codacy.com/app/mziccard/gcloud-java/file/2431752990/issues/source?bid=3140968&fileBranchId=3241696#l984)

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java [hunk 3]
```diff
@@ -964,6 +964,88 @@ public static DiskFilter notEquals(DiskField field, long value) {
...
+     * @see <a href="https://github.com/google/re2/wiki/Syntax">RE2</a>
+     */
+    public static NetworkFilter notEquals(NetworkField field, String value) {
+      return new NetworkFilter(checkNotNull(field), ComparisonOperator.NE, checkNotNull(value));
+    }
+
+    /**
+     * Returns a equals filter for the given field and boolean value.
+     */
+    public static NetworkFilter equals(NetworkField field, boolean value) {
```
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java]: Codacy Issue found: [The method name and parameter number are suspiciously close to equals(Object)](https://www.codacy.com/app/mziccard/gcloud-java/file/2431752990/issues/source?bid=3140968&fileBranchId=3241696#l1036)

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java [hunk 4]
```diff
@@ -1336,6 +1395,217 @@ public Operation resize(final DiskId disk, final long sizeGb, OperationOption...
...
+      }
+    };
+  }
+
+  @Override
+  public Page<Subnetwork> listSubnetworks(String region, SubnetworkListOption... options) {
+    return listSubnetworks(region, options(), optionMap(options));
+  }
+
+  private static Page<Subnetwork> listSubnetworks(final String region,
```
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java]: Codacy Issue found: [Avoid unused private methods such as 'listSubnetworks(String,ComputeOptions,Map)'.](https://www.codacy.com/app/mziccard/gcloud-java/file/2431752993/issues/source?bid=3140968&fileBranchId=3241696#l1449)

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java [hunk 5]
```diff
@@ -1336,6 +1395,217 @@ public Operation resize(final DiskId disk, final long sizeGb, OperationOption...
...
+      throw ComputeException.translateAndThrow(e);
+    }
+  }
+
+  @Override
+  public Page<Network> listNetworks(NetworkListOption... options) {
+    return listNetworks(options(), optionMap(options));
+  }
+
+  private static Page<Network> listNetworks(final ComputeOptions serviceOptions,
```
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java]: Codacy Issue found: [Avoid unused private methods such as 'listNetworks(ComputeOptions,Map)'.](https://www.codacy.com/app/mziccard/gcloud-java/file/2431752993/issues/source?bid=3140968&fileBranchId=3241696#l1558)

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java [hunk 6]
```diff
@@ -1336,6 +1395,217 @@ public Operation resize(final DiskId disk, final long sizeGb, OperationOption...
...
+      throw ComputeException.translateAndThrow(e);
+    }
+  }
+
+  @Override
+  public Page<Subnetwork> listSubnetworks(SubnetworkAggregatedListOption... options) {
+    return listSubnetworks(options(), optionMap(options));
+  }
+
+  private static Page<Subnetwork> listSubnetworks(final ComputeOptions serviceOptions,
```
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java]: Codacy Issue found: [Avoid unused private methods such as 'listSubnetworks(ComputeOptions,Map)'.](https://www.codacy.com/app/mziccard/gcloud-java/file/2431752993/issues/source?bid=3140968&fileBranchId=3241696#l1477)

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java [hunk 7]
```diff
@@ -2066,4 +2318,83 @@ Operation deprecate(ImageId image, DeprecationStatus<ImageId> deprecationStatus,
...
+   * @throws ComputeException upon failure
+   */
+  Subnetwork get(SubnetworkId subnetworkId, SubnetworkOption... options);
+
+  /**
+   * Lists subnetworks for the provided region.
+   *
+   * @throws ComputeException upon failure
+   */
+  Page<Subnetwork> listSubnetworks(String project, SubnetworkListOption... options);
```
Comment (aozarov) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java]: s/project/region/

**Pull request merged**